packages/editor/src/components/block-settings-menu/content-only-settings-menu.js

```diff
@@ -19,53 +20,84 @@ import { store as editorStore } from '../../store';
 import { unlock } from '../../lock-unlock';
 
 function ContentOnlySettingsMenuItems( { clientId, onClose } ) {
-	const { entity, onNavigateToEntityRecord } = useSelect(
-		( select ) => {
-			const {
-				getBlockEditingMode,
-				getBlockParentsByBlockName,
-				getSettings,
-				getBlockAttributes,
-			} = select( blockEditorStore );
-			const contentOnly =
-				getBlockEditingMode( clientId ) === 'contentOnly';
-			if ( ! contentOnly ) {
-				return {};
-			}
-			const patternParent = getBlockParentsByBlockName(
-				clientId,
-				'core/block',
-				true
-			)[ 0 ];
-
-			let record;
-			if ( patternParent ) {
-				record = select( coreStore ).getEntityRecord(
-					'postType',
-					'wp_block',
-					getBlockAttributes( patternParent ).ref
+	const { entity, onNavigateToEntityRecord, canEditTemplateParts } =
+		useSelect(
+			( select ) => {
+				const {
+					getBlockEditingMode,
+					getBlockParentsByBlockName,
+					getSettings,
+					getBlockAttributes,
+				} = select( blockEditorStore );
+				const contentOnly =
+					getBlockEditingMode( clientId ) === 'contentOnly';
+				const _canEditTemplateParts = select( coreStore ).canUser(
+					'create',
+					'templates'
 				);
-			} else {
-				const { getCurrentPostType, getCurrentTemplateId } =
-					select( editorStore );
-				const currentPostType = getCurrentPostType();
-				const templateId = getCurrentTemplateId();
-				if ( currentPostType === 'page' && templateId ) {
+				if ( ! contentOnly ) {
+					return {
+						canEditTemplateParts: _canEditTemplateParts,
```

Comment:
I don't think this is needed in the early return (see the later comment).

Reply:
Ah yeah, good call!